### PR TITLE
Removed virtual from data contracts

### LINKED ISSUE #344: Use struct for data-contracts
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description
Updating the data contract to match the SDK .NET standards. This removes virtual from all the properties which will block the ability to mock the contracts.

## Type of change

Please delete options that are not relevant.

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Closing issues

Put closes #344

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosConflictSettings.cs [hunk 1]
```diff
@@ -34,22 +34,22 @@ public class CosmosConflictSettings
...
-        public virtual OperationKind OperationKind { get; internal set; }
+        public OperationKind OperationKind { get; internal set; }
 
         /// <summary>
         /// Gets the content of the Conflict resource in the Azure Cosmos DB service.
         /// </summary>
         /// <typeparam name="T">The type to use to deserialize the content.</typeparam>
         /// <param name="cosmosJsonSerializer">(Optional) <see cref="CosmosJsonSerializer"/> to use while parsing the content.</param>
         /// <returns>An instance of T</returns>
-        public virtual T GetResource<T>(CosmosJsonSerializer cosmosJsonSerializer = null)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosConflictSettings.cs]: Virtual on methods should be fine right?

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosDatabaseSettings.cs [hunk 2]
```diff
@@ -60,13 +60,24 @@ namespace Microsoft.Azure.Cosmos
...
         /// </summary>
         public CosmosDatabaseSettings()
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CosmosDatabaseSettings"/> class for the Azure Cosmos DB service.
+        /// </summary>
+        /// <param name="id">The Id of the resource in the Azure Cosmos service.</param>
+        public CosmosDatabaseSettings(string id)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosDatabaseSettings.cs]: Is this for convenience?

Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosDatabaseSettings.cs]: Yes

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosConflictSettings.cs (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosConflictSettings.cs]: What is the point of the method's being virtual if nothing else can be mocked?

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosConflictSettings.cs [hunk 3]
```diff
@@ -34,22 +34,22 @@ public class CosmosConflictSettings
...
+        public T GetResource<T>(CosmosJsonSerializer cosmosJsonSerializer = null)
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosConflictSettings.cs]: Use default json serializer for conflicts.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosConflictSettings.cs (hunk 1 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosConflictSettings.cs]: It should be internal. @ealsur why is it public?

Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosConflictSettings.cs]: Actually its not required. Dead code.

Comment (ealsur) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/CosmosConflictSettings.cs]: Yeah, this is dead code, the correct one is `ReadConflictContent`

**Pull request merged**